org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/refactoring2/participant/ResourceRelocationProcessor.xtend

```diff
@@ -66,26 +67,40 @@ class ResourceRelocationProcessor {
 					IProgressMonitor pm) throws CoreException, OperationCanceledException {
 		if (uriChanges.empty)
 			return null
+
+		val subMonitor = SubMonitor.convert(pm)
+		// declaring the task and its effort in 'SubMonitor.convert(...)' doesn't yield the expected UI updates
+		//  so let's do it separately; the total effort of '5' is chosen for weighting the subseqent efforts
+		subMonitor.beginTask('Preparing the refactoring...', 5)
+		
 		val changeSerializer = changeSerializerProvider.get();
 		val resourceSet = resourceSetProvider.get(project)
 		liveScopeResourceSetInitializer.initialize(resourceSet)
 		val context = new ResourceRelocationContext(type, uriChanges, issues, changeSerializer, resourceSet)
-		executeParticipants(context)
+		executeParticipants(context, subMonitor.split(1)) // requires effort of loading affected resources
+		
 		val changeConverter = changeConverterFactory.create(name, [
 			(!(it instanceof MoveResourceChange || it instanceof RenameResourceChange)
 				|| !excludedResources.contains(modifiedElement))
 		], issues)
+		
+		val modificationApplicationMonitor = subMonitor.split(4) // remaining effort is assigned to 'changeSerializer's work
+		changeSerializer.progressMonitor = modificationApplicationMonitor
 		changeSerializer.applyModifications(changeConverter)
+		modificationApplicationMonitor.done
 		return changeConverter.change
 	}
 
-	protected def void executeParticipants(ResourceRelocationContext context) {
+	protected def void executeParticipants(ResourceRelocationContext context, SubMonitor monitor) {
```

Comment:
see also @kthoms comment in the other PR. It would be more idiomatic to use IProgressMonitor in the method signature.

Reply:
Maybe, but in IMHO here it would just unnecessary complexity.